website/pages/docs/connect/observability/ui-visualization.mdx

```diff
@@ -0,0 +1,501 @@
+will be allowed access to the metrics through the proxy.
+
+Even with ACLs enabled, the proxy endpoint doesn't deeply understand the query
+language of the backend so there is no way it can enforce least-privilege access
+to only specific service-related metrics.
+
+_If you are not comfortable with all users of Consul having full access to the
+metrics backend, you should not use the proxy and find an alternative like using
+a custom provider that can query the metrics backend directly_.
+
+##### Path Allowlist
+
+To limit exposure of the metrics backend, paths must be explicitly added to an
+allowlist to avoid exposing unintended parts of the API. For example with
+Prometheus, both the `/api/v1/query_range` and `/api/v1/query` endpoints are
+needed to load time-series and individual stats. If the proxy had the `base_url`
+set to `http://prometheus-server` then the proxy would also expose read access
+to several other endpoints such as `/api/v1/status/config` which includes all
+Prometheus configuration which might include sensitive information.
+
+The default value for `metrics_proxy.path_allowlist` is `["/api/v1/query_range",
+"/api/v1/query"]` so that if you are using the built in provider, this is
+already limited to the essential endpoints.
+
+If you use a custom provider that uses the metrics proxy, you'll need to
+explicitly set the allowlist based on the endpoints the provider needs to
+access.
+
+#### Adding Headers
+
+It is also possible to configure the proxy to add one or more headers to
+requests as they pass through. This is useful when the metrics backend requires
+authentication. For example if your metrics are shipped to a hosted provider,
+you could provision an API token specifically for the Consul UI and configure
+the proxy to add it as in the example below. This keeps the API token only
+visible to Consul operators in the configuration file while UI users can query
+the metrics they need without separately obtaining a token for that provider or
+having a token exposed to them that they might be able to use elsewhere.
+
+```hcl
+ui_config {
+  enabled = true
+  metrics_provider = "example-apm"
+  metrics_proxy {
+    base_url = "http://example-apm/api/v1/metrics"
+    add_headers = [
+      {
+        name = "Authorization"
+        value = "Bearer <token>"
+      }
+    ]
+  }
+}
+```
+
+## Custom Metrics Providers
+
+Consul 1.9.0 includes a built-in provider for fetching metrics from
+[Prometheus](https://prometheus.io). To enable others to use the feature along
+with other existing metrics stores and hosted services, we created a "metrics
+provider" interface in Javascript. A custom provider may be written and the
+javascript file served by the Consul agent.
+
+~> **Note**: this interface is _experimental_ and may change in breaking ways or
+ be removed entirely as we discover the needs of the community. Please provide
+ feedback on Github or [Discuss](https://discuss.hashicorp.com/) on how you'd
```

Comment:
Small nit: "Github" should be "GitHub". Maybe it should also include a link like Discuss?